pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -375,6 +434,16 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
 
         return checkpoint
 
+    def get_loops_state_dict(self):
+        return {"result_collections": self.get_result_collections_state_dict()}
+
+    def get_result_collections_state_dict(self):
+        return {
+            RunningStage.TRAINING.value: self.trainer.train_loop.results.state_dict(),
+            RunningStage.VALIDATING.value: self.trainer.evaluation_loop._val_results.state_dict(),
+            RunningStage.TESTING.value: self.trainer.evaluation_loop._test_results.state_dict(),
```

Comment:
should we consider a state_dict() method on the loops?
If a loop has a child-loop, it can collect its state dict as well. 
not sure, maybe not worth it.

Reply:
I think it would make sense when progress tracking is implemented on all loops.